### DISCUSSION TITLE
creating a hello triangle standalone sample

### DISCUSSION THREAD

**Discussion by GokulDas027** [google/filament#5232] [Q&A] [answered]
trying to create a standalone hello triangle sample,
* does filament creates a window on it's own, or need to use ImGui?
* is there an api to load .filamat material from file? (couldn't find in Material and Filamat::MaterialBuilder)
* how to detect user input while in renderloop

I'm referring to doxygen docs and this discussion section for reference, is there any other resources i'm missing.

**Comment by romainguy** [comment, marked as answer]
Window creation and input management is out of the scope of Filament. You'll have to use something like SDL2 or glfw for that.

**Reply by GokulDas027** [reply]
so, in the provided samples, FilamentApp is the helper for handling these, right?

**Reply by romainguy** [reply]
That's correct. And it uses SDL2.

**Comment by romainguy** [comment]
There is not API to load a filamat from a file, your app is supposed to do this. Filament expects materials as a raw pointer to memory.

**Reply by GokulDas027** [reply]
seems it's same for Texture loading but loader is provided for glTF in gltfio.
